bundles/org.openhab.ui/web/src/components/pagedesigner/sitemap/dslUtil.js

```diff
@@ -4,18 +4,30 @@ function writeWidget (widget, indent) {
   if (widget.config) {
     for (let key in widget.config) {
       if (!widget.config[key]) continue
-      dsl += ` ${key}=`
-      if (key === 'item' || Number.isFinite(widget.config[key])) {
-        dsl += widget.config[key]
-      } else if (key === 'mappings') {
-        dsl += '['
-        const mappingsDsl = widget.config.mappings.map((m) =>
-          `${m.split('=')[0]}="${m.substring(m.indexOf('=') + 1)}"`
-        )
-        dsl += mappingsDsl.join(',')
-        dsl += ']'
+      if (key === 'switchEnabled') {
+        dsl += ' switchSupport'
+      } else if (key === 'frequency') {
+        dsl += ' sendFrequency=' + widget.config[key]
+      } else if (key === 'forceAsItem') {
+        dsl += ' forceasitem=' + widget.config[key]
       } else {
-        dsl += '"' + widget.config[key] + '"'
+        dsl += ` ${key}=`
+        if (key === 'item' || Number.isFinite(widget.config[key])) {
+          dsl += widget.config[key]
+        } else if (['mappings', 'visibility', 'valuecolor', 'labelcolor', 'iconcolor'].includes(key)) {
+          dsl += '['
+          const arrayDsl = widget.config[key].map((v) => {
+            // Anything after the first comparator that is a string should be in quotes.
+            // Also quote string if no comparator (i.e. fixed labelcolor or valuecolor).
+            let value = v.substring(0, v.search(/[=<>]/))
+            value += v.substring(v.search(/[=<>]/)).replace(/[A-Za-z][A-Za-z0-9 _-]*/, function (x) { return '"' + x + '"' })
+            return `${value}`
```

Comment:
When I try this:

![image](https://user-images.githubusercontent.com/2004147/194148780-dbbf48a5-92ad-489a-a216-ba9ca61a2aa4.png)

it works properly:

```
Text icon="frontdoor" item=HS1DSZDoorSensor_DoorSensor label="Front Door" valuecolor=[=="CLOSED"="blue",=="OPEN"="red"]
```

However when changed to:

![image](https://user-images.githubusercontent.com/2004147/194149106-762f7e42-3e16-4494-a6d0-08cde495d795.png)

which should be valid - https://www.openhab.org/docs/ui/sitemaps.html#label-and-value-colors - _"If an operator is not specified, the operator will default to =="_, the generated code becomes:

```
Text icon="frontdoor" item=HS1DSZDoorSensor_DoorSensor label="Front Door" valuecolor=[CLOSED=""blue"",OPEN=""red""]
```
and parsing chokes on the repeated double quotes around the colors.

The same seems to happen when an item name is specified in the designer e.g. `HS1DSZDoorSensor_LastChange=="Uninitialized"` - extra double quotes are added:
```
        Text icon="frontdoor" item=HS1DSZDoorSensor_DoorSensor label="Front Door" visibility=[HS1DSZDoorSensor_LastChange==""Uninitialized""]
```

In fact the examples in the sitemaps docs should probably all be tested & working.

Reply:
I did one correction, and for me it now works on all examples that are in the documentation on the website. I did have to submit a small PR to correct a syntax error in one of the examples. Can you check again?

Reply:
It seems there are still some cases where the quotes aren't right (the color names don't have their quotes anymore) :)
(in fact I'd like to ask, why not simply copy what the user is providing verbatim?)

![image](https://user-images.githubusercontent.com/2004147/194648642-504cfb25-6898-483b-a9f8-145cfe29df31.png)

outputs:

```
visibility=[Day_Time=="Morning",Temperature>19]
labelcolor=[Temperature==22="green",Last_Update=="Uninitialized"=gray,LastUpdate=="OPEN"=red,22="green","gray"]
```

Reply:
This was the original code, only for mapping:
```
      } else if (key === 'mappings') {
        dsl += '['
        const mappingsDsl = widget.config.mappings.map((m) =>
          `${m.split('=')[0]}="${m.substring(m.indexOf('=') + 1)}"`
        )
        dsl += mappingsDsl.join(',')
```
Notice it also adds quotes as they come without quotes from the widget.config. I tried generalizing for all, but that makes it more complex. Not sure if it works without these gymnastics.
Anyway, I thing the missing element is to do the quoting not just once, but for all instances of strings, so just missing a 'g' in the search.